crypto/src/key/signature/mod.rs

```diff
@@ -53,13 +60,18 @@ impl Decode for Signature {
         let sig_kind: Option<SignatureKind> = num::FromPrimitive::from_u8(sig_kind_raw);
         let sig_kind =
             sig_kind.ok_or_else(|| serialization::Error::from("Invalid/Unknown signature kind"))?;
-        let data = Vec::decode(input)?;
 
         match sig_kind {
+            SignatureKind::Secp256k1Schnorr => {
+                let data = <[u8; secp256k1::constants::SCHNORR_SIGNATURE_SIZE]>::decode(input)?;
+                let sig = secp256k1::schnorr::Signature::from_slice(&data)
+                    .map_err(|_| serialization::Error::from("Signature deserialization failed"))?;
+                Ok(Signature::Secp256k1Schnorr(sig))
+            }
             SignatureKind::RistrettoSchnorr => {
-                let sig = schnorrkel::Signature::from_bytes(&data).map_err(|_| {
-                    serialization::Error::from("Private Key deserialization failed")
-                })?;
+                let data = Vec::decode(input)?;
+                let sig = schnorrkel::Signature::from_bytes(&data)
+                    .map_err(|_| serialization::Error::from("Signature deserialization failed"))?;
                 Ok(Signature::RistrettoSchnorr(sig))
```

Comment:
There seems to be a bit of an inconsistency. When decoding `Secp256k1`, the payload is supposed to be a fixed number of bytes while `Ristrertto` uses a variable length encoding (based on `Vec`). Is that intentional? If new key types are added while the network is in operation, how is that supposed to be handled?